builtins/amp-ad.js

```diff
@@ -134,8 +140,38 @@ export function installAd(win) {
       }
     }
 
+    /**
+     * @override
+     */
+    onLayoutMeasure() {
+      this.isInFixedContainer_ = this.isPositionFixed();
+    }
+
+    /**
+     * @return {boolean} whether this element or its ancestors have position
+     * fixed (unless they are POSITION_FIXED_TAG_WHITELIST).
+     * This should only be called when a layout on the page was just forced
+     * anyway.
+     */
+    isPositionFixed() {
+      let el = this.element;
+      let body = el.ownerDocument.body;
+      do {
+        if (!POSITION_FIXED_TAG_WHITELIST[el.tagName] &&
```

Comment:
You probably want to exit as soon as you hit a POSITION_FIXED_TAG_WHITELIST parent.